pkg/store/flag_test.go

```diff
@@ -0,0 +1,36 @@
+package store_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/improbable-eng/thanos/pkg/store"
+	"github.com/improbable-eng/thanos/pkg/testutil"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func TestTimeOrDurationValue(t *testing.T) {
+	cmd := kingpin.New("test", "test")
```

Comment:
You don't need kingpin for this test IMO

Reply:
@bwplotka but I really like it :P 

I think it's and e2e test for flag parsing and as the package imports kingpin and has a special function for it, IMO it makes sense.